src/build.d

```diff
@@ -731,16 +758,23 @@ auto sourceFiles()
                 .map!(e => e.name)
                 .filter!(e => !lexerRootFiles.canFind(e.baseName.stripExtension))
                 .array,
-        backend:
-            dirEntries(env["C"], "*.d", SpanMode.shallow)
-                .map!(e => e.name)
-                .filter!(e => !e.baseName.among("dt.d", "obj.d"))
-                .array,
-        backendHeaders: [
-            // can't be built with -betterC
-            "dt",
-            "obj",
-        ].map!(e => env["C"].buildPath(e ~ ".d")).array,
+        backend: ("
+            bcomplex.d evalu8.d divcoeff.d dvec.d go.d gsroa.d glocal.d gdag.d gother.d gflow.d
+            out.d
+            gloop.d compress.d cgelem.d cgcs.d ee.d cod4.d cod5.d nteh.d blockopt.d mem.d cg.d cgreg.d
+            dtype.d debugprint.d fp.d symbol.d elem.d dcode.d cgsched.d cg87.d cgxmm.d cgcod.d cod1.d cod2.d
+            cod3.d cv8.d dcgcv.d pdata.d util2.d var.d md5.d backconfig.d ph2.d drtlsym.d dwarfeh.d ptrntab.d
+            dvarstats.d dwarfdbginf.d cgen.d os.d goh.d barray.d cgcse.d elpicpie.d
+            ".split
+            ~ ( (env["OS"] == "osx") ? ["machobj.d"] : ["elfobj.d"] )
```

Comment:
I presume you copied this from the posix makefil, but as there's windows too shouldn't it be sth. like:

```
( (env["OS"] == "osx") ? ["machobj.d"] : [] )
( (env["OS"] == "linux" || env["OS"] == "freebsd") ? ["elfobj.d"] : [] )
```

Though you can skip all of this logic, because these files have `version (OS)` headers, so there's no problem in keeping things simple and including all.